app/controllers/polls_controller.rb

```
@@ -1,11 +1,12 @@
 class PollsController < ApplicationController
-  
   def index
     @owned_polls = current_user.own_polls
+    # authorize! @owned_polls, to: :show?
   end
 
   def show
-    @pol = current_user.own_polls.find_by(id: params[:id])
+    @poll = current_user.own_polls.find_by(id: params[:id])
```

Let it throw `ActiveRecord::RecordNotFound` exception:

```ruby
@poll = current_user.own_polls.find(params[:id])
```

And then add this code in the header to catch it:

```ruby
class PollsController < ApplicationController
  rescue_from ActiveRecord::RecordNotFound do
    redirect_to polls_path, error: "Poll not found"
  end
```

(I don't remember if that's "error" or "alert" in your project, sorry.